src/views/search/search.jsx

```diff
@@ -95,6 +98,9 @@ var Search = injectIntl(React.createClass({
         }
         return allTab;
     },
+    setSearchTerm: function (searchTerm) {
+        this.props.dispatch(navigationDetails.setSearchTerm(searchTerm));
```

Comment:
It doesn't look like `setSearchTerm` is being called anywhere, so the dispatch won't happen right now. Since this is a one line method, I think you can get rid of this function `setSearchTerm`, and just call `this.props.dispatch(navigationDetails.setSearchTerm(this.props.searchTerm));` above in `componentDidMount()`